framework/contracts/account/manager/src/commands.rs

```diff
@@ -808,22 +779,43 @@ pub fn update_internal_config(deps: DepsMut, info: MessageInfo, config: Binary)
     }
 }
 
+fn query_ownership(deps: Deps, maybe_manager: Addr) -> ManagerResult<String> {
+    let Ownership::<String> { owner, .. } = deps
+        .querier
+        .query_wasm_smart(maybe_manager, &QueryMsg::Ownership {})?;
+
+    owner.ok_or(ManagerError::Admin(cw_controllers::AdminError::NotAdmin {}))
+}
+
 fn assert_admin_right(deps: Deps, sender: &Addr) -> ManagerResult<()> {
     let ownership_test = cw_ownable::assert_owner(deps.storage, sender);
 
+    if ownership_test.is_ok() {
+        return Ok(());
+    }
+
     let account_info = INFO.load(deps.storage)?;
     match account_info.governance_details {
         // If the account has SubAccountMonarch governance, the owner of the monarch of this account also has admin rights over this account
-        GovernanceDetails::SubAccountMonarchy { monarch } => {
-            // We try to query the ownership of the monarch account
-            let Ownership::<String> { owner, .. } = deps
-                .querier
-                .query_wasm_smart(monarch, &QueryMsg::Ownership {})?;
-            // We assert that the sender is the owner
-            if *sender != owner.unwrap() {
-                return Err(ManagerError::Admin(cw_controllers::AdminError::NotAdmin {}));
+        GovernanceDetails::SubAccount {
+            owner_account_manager,
```

Comment:
hmm this is confusing. 

`owner_account_manager` -> The owner of the account manager (which?)
OR 
`owner_account_manager` -> The manager of the account that owns this account (whit we want) 

Maybe there's a better name to apply here

Reply:
I can replace that with `manager` only, because that's what it indicates --> This is a subaccount, managed by an account manager, which we should be able to query an owner of